src/hammer-vlsi/synthesis/yosys/__init__.py

```diff
@@ -0,0 +1,253 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+#  Yosys synthesis plugin for Hammer (as part of OpenROAD installation).
+#
+#  See LICENSE for licence details.
+
+from typing import List, Optional, Dict, Any
+
+import os
+import shutil
+from textwrap import dedent as dd
+
+from hammer_vlsi import HammerSynthesisTool, HammerToolStep, \
+                        PlacementConstraintType, TimeValue, \
+                        HasSDCSupport, TCLTool
+import hammer_tech
+
+
+class YosysSynth(HammerSynthesisTool, HasSDCSupport, TCLTool):
```

Comment:
High level question - can this be made to work with "vanilla" Yosys, or does it require OpenROAD-Yosys? What are the differences?

Reply:
right now i'm focusing on getting the complete OpenROAD flow to work without modification. trying to support yosys separately will take a bit of work and is not a priority right now

Reply:
I understand. Would it be possible for you to perhaps enlighten us on the differences between "OpenROAD Yosys" and vanilla Yosys?

Reply:
nothing is different. openroad has a local install of yosys. they just have a bunch of scripts and setup around it.

Reply:
OpenRoad actually does have their own fork of yosys. Why they do not contribute their changes back to upstream only they know: https://github.com/The-OpenROAD-Project/yosys

Reply:
```
Physical yosys aims at improving the timing optimizaions of yosys/ABC through adding physical-awareness (along with other features) in the synthesis flow. Physical yosys in implemented by Scale Lab at Brown University as part of the OPENROAD project.
```
I wonder if it's backwards-compatible with "regular" yosys.